services/redis.yaml

```diff
@@ -0,0 +1,81 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: sd-redis
+  labels:
+    name: redis
+    app: sd-redis
+spec:
+  ports:
+  - port: 6379
+    name: master
+  clusterIP: None
+  selector:
+    name: redis
+---
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: redis
+spec:
+  serviceName: "sd-redis"
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: sd-redis
+        name: redis
+        role: master
+      name: redis
+    spec:
+      initContainers:
+        - name: config-setter
+          image: busybox
+          env:
+            - name: REDIS_MASTER_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: redis-secret
+                  key: password
+          command: [
+            'sh', '-c',
+            'cp /redis-config-template/redis-master.conf /redis-conf/; sed -i "s/%master-password%/${REDIS_MASTER_PASSWORD}/" /redis-conf/redis-master.conf'
```

Comment:
Main container is a redis container, to expose sed in it, we will have to build a custom container with sed. With init container approach we can isolate any custom conf processing inside init container, and main container will just execute our service.